lib/utils.js

```diff
@@ -262,6 +262,31 @@ exports.setCharset = function setCharset(type, charset) {
   return contentType.format(parsed);
 };
 
+/**
+ * Extend an object
+ *
+ * @param {Object} object The destination object
+ * @param {...Object} source The source objects
+ * @returns {Object}
+ * @private
+ */
+
+exports.extendObject = function(target, source) {
```

Comment:
Instead of adding more code here, can the dependency `utils-merge` we already have perform this function? Or if not, what's the difference between this code and `utils-merge`?

Reply:
Hi @dougwilson, sorry I didn't see you required a merge util, I was scanning for an extending `extend()`  function or corresponding library 🤐. This passes my tests:
```
this.settings[setting] = merge(merge({}, this.settings[setting]), val)
```